### ISSUE TITLE
Comparison with SQLite Plugin 2?

### ISSUE THREAD

**Issue by brody2consult** [brody2consult/cordova-sqlite-storage#547]
From a quick test https://github.com/nolanlawson/cordova-plugin-sqlite-2 seems to do better in some but not all tests in https://github.com/brodybits/Cordova-sqlite-perftest (with 3 minor changes). SQLite plugin 2 claims to be more compliant to the Web SQL spec and seems to be mostly tested against PouchDB. SQLite plugin 2 also uses https://github.com/nolanlawson/node-websql which could let programmers develop and test Web SQL code using NodeJS before testing on a mobile platform.

A minor thing I discovered is that if an integer number is stored in a column with no type specified the value is not read as an integer on Android. I had to change `!==` to `!=` in a couple places in Cordova-sqlite-perftest to get it to work on Android.

SQLite plugin 2 is of course missing the utility close/delete functionality and does not allow the database location to be specified on iOS.

Another thing is that looking at the native platform code, I am not sure how much if it is really original and how much was copied from this project. The iOS version of this project was written by someone else and released under the MIT license while sqlite plugin 2 is released under the Apache 2.0 license.

An idea may be to use the JavaScript from sqlite plugin 2 (which mostly packages node-websql) with the native platform code from this project to support Android/iOS/Windows all together.

_P.S. An additional item to test is the maximum amount of data can be inserted in a single transaction._

**Comment by chauthai** [comment]
As stated in the [blog post for SQLite Plugin 2](https://nolanlawson.com/2016/04/10/introducing-the-cordova-sqlite-plugin-2/) it's a fork of this project.

I like your idea for merging both projects! What's your rough estimate how long a merge might take?

**Comment by brody2consult** [comment]
> As stated in the blog post for SQLite Plugin 2 it's a fork of this project.

Yes it is a fork but from the description it looks more like a rewrite.

> I like your idea for merging both projects! What's your rough estimate how long a merge might take?

Answer in https://github.com/litehelpers/Cordova-sqlite-storage/issues/548#issuecomment-248526954

**Comment by brody2consult** [comment]
Also sqlite-plugin-2 is that it shares the Web SQL API implementation with [nolanlawson / node-websql](https://github.com/nolanlawson/node-websql).

As described in litehelpers/Cordova-sqlite-help#11 developers can use  [nolanlawson / node-websql](https://github.com/nolanlawson/node-websql) directly when porting their apps to Node.js based frameworks such as electron or nw.js (nee node-webkit).
